### ISSUE TITLE
Multiple raster maps same HTML, values on mousover only first map

### ISSUE THREAD

**Issue by ghTaranto** [r-spatial/mapview#402]
Hello, 

when you have multiple maps on the same html page, raster values on mousehover are showed always on the first map. 

This behavior is visible on the [example page](https://r-spatial.github.io/mapview/articles/articles/mapview_02-advanced.html#adjusting-color-col-regions-at-1) where the raster layer poppondorf appears in two maps. On the first map raster values are showed correctly when the cursor is moved on it; when the cursor is moved on the same raster layer in the second map, raster values are showed only in the first map.

Here I reproduced two interactive maps with the same behavior [rpubs example](https://rpubs.com/ghtaranto/835655)

**Comment by ghTaranto** [comment]
> 
> 
> When used in a shiny-app no values show at all and there are some errors in the browser console:
> 
> > leafem: Failed to find window.data
> > Uncaught TypeError: layer.value.toFixed is not a function
> 
> App to reproduce:
> 
> ```
> library(mapview)
> library(raster)
> library(shiny)
> library(leaflet)
> 
> r <- system.file("external/test.grd", package="raster")
> r <- raster(r)
> 
> 
> ui <- fluidPage(
>   mapviewOutput("map"),
>   mapviewOutput("map1")
> )
> 
> server <- function(input, output, session) {
>   output$map <- renderLeaflet({
>     mapview(r)@map
>   })
>   output$map1 <- renderLeaflet({
>     mapview(r)@map
>   })
> }
> shinyApp(ui, server)
> ```

Your code works on my computer. You can use the same raster as in the mapview example:

library(mapview)
library(raster)
library(shiny)
library(leaflet)
**library(plainview)**

ui <- fluidPage(
  mapviewOutput("map"),
  mapviewOutput("map1")
)


server <- function(input, output, session) {
  output$map <- renderLeaflet({
    mapview(**poppendorf[[5]]**)@map
  })
  output$map1 <- renderLeaflet({
    mapview(**poppendorf[[5]]**)@map
  })
}

shinyApp(ui, server)

**Comment by trafficonese** [comment]
Intersting, your code doesnt work at all on mine with the latest github and CRAN version.

I am getting:

> Error: no slot "Map" for this object ob class "mapview"

**Comment by ghTaranto** [comment]
> 
> 
> Interesting, your code doesnt work at all on mine with the latest github and CRAN version.
> 
> I am getting:
> 
> > Error: no slot "Map" for this object ob class "mapview"

Very strange. Can I share some more information with you to figure out why? 

Anyway I have realized that if I change the `layer.name` the mouseover event is correctly showed [rpubs updated](https://rpubs.com/ghtaranto/835655)

 Do you have any idea why?

**Comment by trafficonese** [comment]
Yes, the underlying package (`leafem`) and the Javascript library use the `layerId` (or in mapviews case the `layer.name`) to assign and receive the data to/from the browser window.

But it's still weird, that it is affected by multiple maps. Maybe we could/should use `map.id + layerId` to be more precise? What do you say @tim-salabim ?

**Comment by ghTaranto** [comment]
My knowledge of java and html is quite poor. 

However, what it seems to happen is that every map is named after "layer.name". When the `addImageQuery` method is called its output is showed on the map having "layer.name". If more than one map has the same name, the output is showed only on the first one.

This same result is not happening with the geographic coordinates `mousemove` event, that are correctly displayed on every map. 

I have tried hard to find a workaround with little luck. Hope you can see and point out an easy solution.
